src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/CreateCohortController.cs

```diff
@@ -35,20 +57,55 @@ public IActionResult Index(IndexRequest request)
         [Route("select-provider")]
         public IActionResult SelectProvider(SelectProviderRequest request)
         {
-            var viewModel = new SelectProviderViewModel();//todo: from mapper
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("Error", "Error", new { StatusCode = 400 });
+            }
+
+            var viewModel = _selectProviderViewModelMapper.Map(request);
 
             return View(viewModel);
         }
 
         [Route("select-provider")]
         [HttpPost]
-        public IActionResult SelectProvider(SelectProviderViewModel request)
+        public async Task<IActionResult> SelectProvider(SelectProviderViewModel request)
         {
-            //todo:hit api
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("Error", "Error", new { StatusCode = 400 });
```

Comment:
We would want to just re-render the view if the ModelState is not valid, not redirect to an error page

Reply:
This is done in #29